Tasks/Common/azure-arm-rest/azure-arm-storage.ts

```diff
@@ -0,0 +1,81 @@
+import msRestAzure = require("./azure-arm-common");
+import azureServiceClient = require("./AzureServiceClient");
+import tl = require('vsts-task-lib/task');
+import Q = require('q');
+import util = require("util");
+
+export class StorageManagementClient extends azureServiceClient.ServiceClient {
+    constructor(credentials: msRestAzure.ApplicationTokenCredentials, subscriptionId: string) {
+        super(credentials, subscriptionId);
+
+        this.apiVersion = '2017-06-01';
+        this.acceptLanguage = 'en-US';
+        this.generateClientRequestId = true;
+    }
+
+    public getResourceGroupName(storageAccountName: string): Q.Promise<string> {
+        var httpRequest = new azureServiceClient.WebRequest();
+        httpRequest.method = 'GET';
+        // Getting all storage accounts (along with resource group names) for the given subscription.
+        httpRequest.uri = this.getRequestUri(
+            '//subscriptions/{subscriptionId}/providers/Microsoft.Storage/storageAccounts',
+            {}
+        );
+        var resourceGroupName: string = "";
+        var deferred = Q.defer<string>();
+        this.beginRequest(httpRequest).then(function(result) {
+            var httpResponse: azureServiceClient.WebResponse = result;
+            var response = httpResponse.body["value"];
+            for(let i = 0; i<response.length; i++) {
+                var obj = response[i];
```

Comment:
create an interface for storage account in azureModels and use that type for obj.